controllers/integrationpipeline/integrationpipeline_controller.go

```diff
@@ -109,14 +109,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	return controller.ReconcileHandler([]controller.Operation{
 		adapter.EnsureStatusReportedInSnapshot,
-		adapter.EnsureStatusReported,
 		adapter.EnsureEphemeralEnvironmentsCleanedUp,
 	})
 }
 
 // AdapterInterface is an interface defining all the operations that should be defined in an Integration adapter.
 type AdapterInterface interface {
-	EnsureStatusReported() (controller.OperationResult, error)
 	EnsureStatusReportedInSnapshot() (controller.OperationResult, error)
```

Comment:
and this function to something like this?
```suggestion
	EnsureIntegrationPipelineRunStatusReportedToSnapshot() (controller.OperationResult, error)
```
or
```suggestion
	EnsurePipelineRunStatusReportedToSnapshot() (controller.OperationResult, error)
```
or something else?

Reply:
This function is added by @MartinBasti, it is not just to update PLR status to snapshot, also include the env provision/SEB deployment error.

Reply:
Don't do unrelated changes in this PR please.

Reply:
I won't update it in this PR. Actually I don't feel we need to update it.